openverse_catalog/dags/data_refresh/dag_factory.py

```diff
@@ -1,135 +1,143 @@
+    GROUP_ID as REFRESH_POPULARITY_METRICS_GROUP_ID,
+)
+from data_refresh.refresh_popularity_metrics_task_factory import (
+    create_refresh_popularity_metrics_task_group,
+)
+from data_refresh.refresh_view_data_task_factory import create_refresh_view_data_task
 
 
 logger = logging.getLogger(__name__)
 
-
-DATA_REFRESH_POOL = "data_refresh"
+REFRESH_MATERIALIZED_VIEW_TASK_ID = operators.UPDATE_DB_VIEW_TASK_ID
+# The first task in the refresh_popularity_metrics TaskGroup
+REFRESH_POPULARITY_METRICS_TASK_ID = (
+    f"{REFRESH_POPULARITY_METRICS_GROUP_ID}"
+    f".{operators.UPDATE_MEDIA_POPULARITY_METRICS_TASK_ID}"
+)
 
 
-def response_filter_data_refresh(response: Response) -> str:
-    """
-    Filter for the `trigger_data_refresh` task, used to grab the endpoint needed
-    to poll for the status of the triggered data refresh. This information will
-    then be available via XCom in the downstream tasks.
+@provide_session
+def _month_check(dag_id: str, session: SASession = None) -> str:
     """
-    status_check_url = response.json()["status_check"]
-    return urlparse(status_check_url).path
+    Checks whether there has been a previous DagRun this month. If so,
+    returns the task_id for the matview refresh task; else, returns the
+    task_id for refresh popularity metrics task.
 
+    Required Arguments:
 
-# Response check to the `wait_for_completion` Sensor. Processes the response to
-# determine whether the task can complete.
-def response_check_wait_for_completion(response: Response) -> bool:
-    """
-    Response check to the `wait_for_completion` Sensor. Processes the response to
-    determine whether the task can complete.
+    dag_id:     id of the currently running Dag
     """
-    data = response.json()
-
-    if data["active"]:
-        # The data refresh is still running. Poll again later.
-        return False
+    # Get the current DagRun
+    DR = DagRun
+    current_dagrun = (
+        session.query(DR).filter(DR.dag_id == dag_id, DR.state == State.RUNNING)
+    ).first()
+
+    # If `force_refresh_metrics` has been passed in the dagrun config, then
+    # immediately return the task_id to refresh popularity metrics without
+    # doing the month check.
+    force_refresh_metrics = current_dagrun.conf.get("force_refresh_metrics")
+    if force_refresh_metrics is not None:
+        logger.info(f"`force_refresh_metrics` is set to {force_refresh_metrics}.")
+        return (
+            REFRESH_POPULARITY_METRICS_TASK_ID
+            if force_refresh_metrics
+            else REFRESH_MATERIALIZED_VIEW_TASK_ID
+        )
 
-    if data["error"]:
-        raise AirflowException("Error triggering data refresh.")
+    # Get the most recent successful dagrun for this Dag
+    DR = DagRun
+    query = (
+        session.query(DR)
+        .filter(DR.dag_id == dag_id, DR.state == State.SUCCESS)
+        .order_by(DR.start_date.desc())
+    )
+    latest_dagrun = query.first()
```

Comment:
Did we want to collapse this into a single statement like the `current_dagrun`?
```suggestion
    latest_dagrun = (
        session.query(DR)
        .filter(DR.dag_id == dag_id, DR.state == State.SUCCESS)
        .order_by(DR.start_date.desc())
        .first()
    )
```